cores/cosa/Cosa/INET/SNMP.cpp

```diff
@@ -34,25 +34,26 @@ const uint8_t SNMP::ARDUINO_MIB_OID[] __PROGMEM = {
 bool
 SNMP::MIB2_SYSTEM::is_request(PDU& pdu)
 {
+  uint8_t mib_baselen = pgm_read_byte(SNMP::MIB2_SYSTEM::OID);
+  if(pdu.oid.length > (mib_baselen + 1)) return (false);
+
   // Match with SNMP MIB-2 System OID root
   int sys = pdu.oid.match(OID);
-  if (sys < 0) return (false);
+  if (sys < -1) return (false);
 
   // Get next value or step to next mib
   if (pdu.type == SNMP::PDU_GET_NEXT) {
+    if (sys < 0) {
```

Comment:
I saw 0 being an exact match of parent MIB - which does indeed lead to sysServices as GET_NEXT MIB but I also want to return sysServices for GET_NEXT for any prior MIB to to root MIB so that is why I let 0 and -1 fall through in my update.

I hope that makes sense - as I said in push request I still think there is tweaking to be done here on this change, I just know these changes make SNMP requests work as intended as I tested them quite extensively and wanted to submit them at least as a starting point.
Cheers